be/src/storage/rowset/column_reader.cpp

```diff
@@ -686,25 +687,32 @@ size_t ColumnReader::mem_usage() const {
     return size;
 }
 
-size_t ColumnReader::_get_gram_num_for_ngram() const {
+NgramBloomFilterReaderOptions ColumnReader::_get_reader_options_for_ngram() const {
     // initialize with invalid number
-    size_t gram_num = 0;
+    NgramBloomFilterReaderOptions reader_options;
     std::shared_ptr<TabletIndex> ngram_bf_index;
 
     Status status = _segment->tablet_schema().get_indexes_for_column(_column_unique_id, NGRAMBF, ngram_bf_index);
     if (!status.ok() || ngram_bf_index.get() == nullptr) {
-        return gram_num;
+        return reader_options;
     }
 
     const std::map<std::string, std::string>& index_properties = ngram_bf_index->index_properties();
```

Comment:
Is index_properties is a persistent string-to-string map? it seems use a bitvector to store some encoded flag is enough, just like in page.flags in Linux or cr0~3 register of intel cpu.